extensions/sovity-messenger/src/main/java/de/sovity/edc/extension/messenger/SovityMessengerRegistry.java

```diff
@@ -89,14 +117,23 @@ public <IN extends SovityMessage> void registerSignal(Class<IN> clazz, String ty
      * @return The function to process this message type.
      */
     @SuppressWarnings("unchecked")
-    public Handler<Object, Object> getHandler(String type) {
-        return (Handler<Object, Object>) handlers.get(type);
+    public HandlerWithClaims<Object, Object> getHandler(String type) {
+        return (HandlerWithClaims<Object, Object>) handlers.get(type);
     }
 
-    private static <IN extends SovityMessage> String getTypeViaIntrospection(Class<IN> incomingMessage) throws NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
+    @SneakyThrows
+    private static <IN extends SovityMessage> String getTypeViaIntrospection(Class<IN> incomingMessage) {
         val defaultConstructor = incomingMessage.getConstructor();
         defaultConstructor.setAccessible(true);
         val type = defaultConstructor.newInstance().getType();
         return type;
     }
+
+    private <IN extends SovityMessage, OUT> void internalRegister(Class<IN> clazz, String type, BiFunction<ClaimToken, IN, OUT> handler) {
```

Comment:
such a name, only to see it does so little, in the end, how about `registerIfNotExists`